tool/tctl/common/auth_command.go

```diff
@@ -644,6 +644,20 @@ func (a *AuthCommand) generateUserKeys(clusterAPI auth.ClientI) error {
 		return trace.Wrap(err)
 	}
 	fmt.Printf("\nThe credentials have been written to %s\n", strings.Join(filesWritten, ", "))
+
+	expires, err := key.TeleportTLSCertValidBefore()
+	if err != nil {
+		logrus.Warnf("Failed to check TTL validity: %v.", trace.Wrap(err))
+		return nil
+	}
+	certRequestTime := time.Now()
+	if certRequestTime.Add(a.genTTL).Sub(expires) > time.Minute {
+		logrus.Warnf("requested ttl of %s was not granted. user may have a role with a shorter max session ttl"+
+			" or an existing session ending before the requested TTL. proceeding with %s",
+			a.genTTL,
+			expires.Sub(certRequestTime).Round(time.Second))
```

Comment:
nit: Capitalize messages? It's fine to do for log statements.

```suggestion
		logrus.Warnf("Requested TTL of %s was not granted. User may have a role with a shorter max session TTL"+
			" or an existing session ending before the requested TTL. Proceeding with %s.",
			a.genTTL,
			expires.Sub(certRequestTime).Round(time.Second))
```

Reply:
Shouldn't we output these messages to stdout or stderr? The user at the console might want to know that there's some discrepancy between the requested and the granted TTLs.